### PR TITLE
Add support for cassandra filtering, update docs

### PR DESCRIPTION
**Pull request by alexleventer**
- Add support for cassandra filtering, update docs

### CODE REVIEW

File: docs/docs/modules/data_connection/vectorstores/integrations/cassandra.mdx [hunk 1]
```diff
@@ -71,5 +72,12 @@ const vectorStore = await CassandraStore.fromTexts(
   cassandraConfig
 );
 
+## Querying docs
```
Comment (jacoblee93) [inline docs/docs/modules/data_connection/vectorstores/integrations/cassandra.mdx:75]: Use `//`

File: docs/docs/modules/data_connection/vectorstores/integrations/cassandra.mdx [hunk 2]
```diff
@@ -71,5 +72,12 @@ const vectorStore = await CassandraStore.fromTexts(
...
+
 const results = await vectorStore.similaritySearch("Green yellow purple", 1);
+
+or filtered query:
```
Comment (jacoblee93) [inline docs/docs/modules/data_connection/vectorstores/integrations/cassandra.mdx:79]: Comment out

File: langchain/src/vectorstores/cassandra.ts [hunk 3]
```diff
@@ -89,17 +98,20 @@ export class CassandraStore extends VectorStore {
...
+    k: number,
+    filter?: this["FilterType"]
   ): Promise<[Document, number][]> {
     if (!this.isInitialized) {
       await this.initialize();
     }
 
-    const queryStr = this.buildSearchQuery(query, k);
+    const queryStr = this.buildSearchQuery(query, k, filter);
+    console.log(queryStr);
```
Comment (jacoblee93) [inline langchain/src/vectorstores/cassandra.ts:114]: Remove log

File: langchain/src/vectorstores/cassandra.ts [hunk 4]
```diff
@@ -89,17 +98,20 @@ export class CassandraStore extends VectorStore {
...
    * @param query The query vector.
    * @param k The number of similar vectors to return.
+   * @param filter
    * @returns Promise that resolves with an array of tuples, each containing a Document and a score.
    */
   async similaritySearchVectorWithScore(
     query: number[],
-    k: number
+    k: number,
+    filter?: this["FilterType"]
```
Comment (jacoblee93) [inline langchain/src/vectorstores/cassandra.ts:107]: You may want to declare an actual type here?

**Review by jacoblee93 — COMMENTED** [review]
Small fixes in docs and a question or two

File: langchain/src/vectorstores/cassandra.ts (hunk 4 above)
Comment (alexleventer) [reply langchain/src/vectorstores/cassandra.ts]: Fixed

File: langchain/src/vectorstores/cassandra.ts (hunk 3 above)
Comment (alexleventer) [reply langchain/src/vectorstores/cassandra.ts]: Fixed

File: docs/docs/modules/data_connection/vectorstores/integrations/cassandra.mdx (hunk 2 above)
Comment (alexleventer) [reply docs/docs/modules/data_connection/vectorstores/integrations/cassandra.mdx:83]: Fixed

File: docs/docs/modules/data_connection/vectorstores/integrations/cassandra.mdx (hunk 1 above)
Comment (alexleventer) [reply docs/docs/modules/data_connection/vectorstores/integrations/cassandra.mdx:77]: Fixed

**Comment by alexleventer** [comment]
Thanks for the review @jacoblee93 ! Everything has been addressed 🙏

File: langchain/secure-connect-test.zip
Comment (jacoblee93) [inline langchain/secure-connect-test.zip:1]: @alexleventer did you leak this accidentally?

File: langchain/src/vectorstores/cassandra.ts [hunk 5]
```diff
@@ -10,12 +10,20 @@ export interface Column {
   name: string;
 }
 
+export interface Index {
+  name: string;
+  value: string;
+}
+
+export type CassandraVectorStoreFilterType = string[];
```
Comment (jacoblee93) [inline langchain/src/vectorstores/cassandra.ts:18]: You will need to `declare` this in your class - see:

https://github.com/langchain-ai/langchainjs/blob/main/langchain/src/vectorstores/analyticdb.ts#L45

**Review by jacoblee93 — CHANGES_REQUESTED** [review]

File: langchain/secure-connect-test.zip
Comment (alexleventer) [reply langchain/secure-connect-test.zip:1]: oops - moved a little too fast. There aren't any creds stored in the zip file. removed

**Comment by jacoblee93** [comment]
Thanks!

**Review by jacoblee93 — APPROVED** [review]

**Pull request merged**